subprojects/model-core/src/main/java/org/gradle/internal/properties/annotations/AbstractTypeMetadataWalker.java

```diff
@@ -169,15 +175,18 @@ protected void walkNestedIterable(Object node, String qualifiedName, BiConsumer<
 
         @Override
         protected void walkNestedChild(Object parent, String childQualifiedName, PropertyMetadata propertyMetadata, InstanceMetadataVisitor visitor, Consumer<Object> handler) {
-            walkNestedChildIfPresentOrElse(
+            boolean isElementOfCollection = false;
+            walkNestedChildIfPresent(
                 () -> getChild(parent, propertyMetadata),
                 childQualifiedName,
+                propertyMetadata,
                 visitor,
-                handler,
-                () -> visitor.visitMissingNested(childQualifiedName, propertyMetadata));
+                isElementOfCollection,
+                handler
+            );
         }
 
-        private static void walkNestedChildIfPresentOrElse(Supplier<Object> unpacker, String qualifiedName, InstanceMetadataVisitor visitor, Consumer<Object> handler, Runnable missingHandler) {
+        private void walkNestedChildIfPresent(Supplier<Object> unpacker, String qualifiedName, PropertyMetadata propertyMetadata, InstanceMetadataVisitor visitor, boolean isElementOfCollection, Consumer<Object> handler) {
```

Comment:
This walks the child even if it's not present. Shall we just call it `walkNestedChild()`?

Reply:
Renamed